Feature/initial ps4 ime keyboard

w1naenator opened the pull request:
Initial system keyboard.
<img width="639" alt="image" src="https://github.com/user-attachments/assets/f0576b9f-e07c-4ec3-b562-cb44e5becbd0" />

Commit:
Add initial on-screen keyboard support (mouse input, works in BB)

Commit:
Fix crash on Shift press.

Commit:
Fix clang-format

polybiusproxy commented on src/core/libraries/ime/ime_dialog_ui.cpp:
```diff
@@ -337,6 +340,93 @@ void ImeDialogUi::DrawMultiLineInputText() {
+    // Fix: safely push/pop style only if shift was enabled before clicking
+    bool highlight = shift_enabled;
+    if (highlight)
+        PushStyleColor(ImGuiCol_Button, ImVec4(0.2f, 0.6f, 1.0f, 1.0f));
```
```diff
-    if (highlight)
+    if (highlight) {
        PushStyleColor(ImGuiCol_Button, ImVec4(0.2f, 0.6f, 1.0f, 1.0f));
+    }
```

polybiusproxy commented on src/core/libraries/ime/ime_dialog_ui.cpp:
```diff
@@ -337,6 +340,93 @@ void ImeDialogUi::DrawMultiLineInputText() {
+    }
+
+    if (highlight)
+        PopStyleColor();
```
```diff
-    if (highlight)
+    if (highlight) {
        PopStyleColor();
+    }
```

polybiusproxy commented on src/core/libraries/ime/ime_dialog_ui.cpp:
```diff
@@ -337,6 +340,93 @@ void ImeDialogUi::DrawMultiLineInputText() {
 
+void ImeDialogUi::DrawKeyboard() {
+    static bool shift_enabled = false;
+    enum class KeyboardMode { Letters, Symbols };
```
Don't define enum types inside of a function, since this will be moved to a common implementation, move this to a header file.

polybiusproxy commented on src/core/libraries/ime/ime_dialog_ui.cpp:
```diff
@@ -337,6 +340,93 @@ void ImeDialogUi::DrawMultiLineInputText() {
+
+    auto draw_row = [&](const char* row, float offset_x) {
+        SetCursorPosX(offset_x);
+        for (int i = 0; row[i]; ++i) {
```
```diff
-        for (int i = 0; row[i]; ++i) {
+        for (int i = 0; row[i] != '\0'; i++) {
```
Wouldn't add implicit checks for these things.

polybiusproxy reviewed (changes requested):
Reviewing what I've noticed so far.

Commit:
Some fixes from @polybiusproxy comments

Commit:
Use vierual keyboard in ImeUi and ImeDialogUi
Added some spam to log.
Some files added, CMakeList modified.

Commit:
Redesign layout closer original

Commit:
CLang fix

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
some font fixes

Commit:
Merge branch 'Feature/initial-ps4-ime-keyboard' of https://github.com/w1naenator/shadPS4 into Feature/initial-ps4-ime-keyboard

Commit:
added controller shortcuts to key configuration

mailwl commented on CMakeSettings.json:
```diff
@@ -6,7 +6,7 @@
       "buildRoot": "${projectDir}\\Build\\${name}",
       "installRoot": "${projectDir}\\Install\\${name}",
-      "cmakeCommandArgs": "",
+      "cmakeCommandArgs": "-DENABLE_QT_GUI=ON -DCMAKE_PREFIX_PATH=C:\\Qt\\6.8.2\\msvc2022_64",
```
remove this change

StevenMiller123 replied:
Given that this is a draft, I'd assume that change was just something temporary they did.

Commit:
Build: Recovered CMakeSettings.json configuration file

Input Handling: Temporarily removed custom keyboard shortcut code; navigation is now handled via ImGui's native navigation system

Virtual Keyboard:

Modified first symbols layout (based on PS5 layout)

Added second symbols layout (based on PS5 layout)

Fonts: Added required symbols to ImGui font atlas for correct rendering in keyboard layouts

Commit:
CLang fixes

Commit:
Added accent letters layouts for english (from PS5)

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
full redesign.

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
Full redesign

Commit:
Merge branch 'Feature/initial-ps4-ime-keyboard' of https://github.com/w1naenator/shadPS4 into Feature/initial-ps4-ime-keyboard

Commit:
syncMerge branch 'Feature/initial-ps4-ime-keyboard' of https://github.com/w1naenator/shadPS4 into Feature/initial-ps4-ime-keyboard

Commit:
Merge branch 'Feature/initial-ps4-ime-keyboard' of https://github.com/w1naenator/shadPS4 into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
replace ImGuiNavInput with ImGuiKey
added repeat event selectable for each key.

Commit:
extfx

Commit:
clang, reuse fixes.

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
Use the colors from OrbisImeParamExtended if they are defined; otherwise, use the default colors.

Commit:
Merge branch 'Feature/initial-ps4-ime-keyboard' of https://github.com/w1naenator/shadPS4 into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
Undertale softlock fix.

Commit:
Merge branch 'Feature/initial-ps4-ime-keyboard' of https://github.com/w1naenator/shadPS4 into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
Initial caret draw/control

Commit:
Merge branch 'Feature/initial-ps4-ime-keyboard' of https://github.com/w1naenator/shadPS4 into Feature/initial-ps4-ime-keyboard

georgemoralis committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

Commit:
Rebase

Commit:
Rebase second try

Commit:
Clang fix

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'shadps4-emu:main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'shadps4-emu:main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'shadps4-emu:main' into Feature/initial-ps4-ime-keyboard

w1naenator committed:
Merge branch 'main' into Feature/initial-ps4-ime-keyboard
